### PR TITLE
[SYCL] Throw for invalid reqd_work_group_size

### PR DESCRIPTION
**Pull request by KornevNikita**
Also fix alignment

### CODE REVIEW

**Comment by jinz2014** [comment]
Is the namespace of `reqd_sub_group_size` still `intel`  (intel::reqd_sub_group_size) ?
Is reqd_work_group_size specific to Intel devices ? 
Thanks.

**Comment by KornevNikita** [comment]
> Is reqd_work_group_size specific to Intel devices ?

Hi, I geuss `reqd_sub_group_size` is still in `intel` namespace. But `reqd_work_group_size` is probably generic, as it is in `sycl` namespace. I'm not familiar with these things, let's maybe ask @steffenlarsen

**Comment by KornevNikita** [comment]
@intel/llvm-reviewers-runtime @intel/sycl-language-enabling-triage take a look please.

File: sycl/test-e2e/Basic/reqd_work_group_size_check_exception.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+
+#define CHECK_INVALID_REQD_WORK_GROUP_SIZE(Dim, ...)                           \
+  {                                                                            \
+    bool ExceptionThorwn = false;                                              \
```
Comment (maarquitos14) [inline sycl/test-e2e/Basic/reqd_work_group_size_check_exception.cpp:10]: Nit: 
```suggestion
    bool ExceptionThrown = false;                                              \
```

Comment (maarquitos14) [reply sycl/test-e2e/Basic/reqd_work_group_size_check_exception.cpp:10]: Also, replace the uses of this variable.

File: sycl/test-e2e/Basic/reqd_work_group_size_check_exception.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+
+#define CHECK_INVALID_REQD_WORK_GROUP_SIZE(Dim, ...)                           \
```
Comment (dm-vodopyanov) [inline sycl/test-e2e/Basic/reqd_work_group_size_check_exception.cpp:8]: nit: rewrite to a function with usage of C++ variadic arguments

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by KornevNikita** [comment]
Disabled the test on hip [6810005](https://github.com/intel/llvm/pull/9237/commits/6810005932b838f5d1f2ed5477084b51c52e9c89). I'll create an issue once it's merged.

**Pull request merged**